### ISSUE TITLE
Can not connect SSH connection failed: Host key is not trusted

### ISSUE THREAD

**Issue by thiagolmoraes** [ronf/asyncssh#364]
Hi, 

Im trying to connect with simple script, but I'm receiving this message ````SSH connection failed: Host key is not trusted````, I already connected and accepted the key. the host is not showing in .ssh/known_hosts

```
import asyncio, asyncssh, sys

async def run_client():
    async with asyncssh.connect('172.18.17.9', username="user", password="admin", port=9321) as conn:
        result = await conn.run('display version', check=True)
        print(result.stdout, end='')

try:
    asyncio.get_event_loop().run_until_complete(run_client())
except (OSError, asyncssh.Error) as exc:
    sys.exit('SSH connection failed: ' + str(exc))
```

**Comment by ronf** [comment]
If the host (or more specifically the IP address of 172.18.17.9) you are trying to connect to is not showing up in .ssh/known_hosts, that would explain the error you're seeing. If you used OpenSSH to connect and accept the key, the question is why it's not showing up. Do you have OpenSSH configured to use a different location for known_hosts?

Out of curiosity, do you have hashing enabled for your known_hosts entries? AsyncSSH should support such entries, but it's possible that OpenSSH may have changed the hashing algorithm at some point. I just did a quick test of that here, though, and it still seems to be working fine for me.

**Comment by thiagolmoraes** [comment]
> If the host (or more specifically the IP address of 172.18.17.9) you are trying to connect to is not showing up in .ssh/known_hosts, that would explain the error you're seeing. If you used OpenSSH to connect and accept the key, the question is why it's not showing up. Do you have OpenSSH configured to use a different location for known_hosts?
> 
> Out of curiosity, do you have hashing enabled for your known_hosts entries? AsyncSSH should support such entries, but it's possible that OpenSSH may have changed the hashing algorithm at some point. I just did a quick test of that here, though, and it still seems to be working fine for me.

I'm using Putty do connect and accept the key, but the the interesting the key is not beeing added in .ssh/known_hosts.

**Comment by ronf** [comment]
PuTTY doesn't write known hosts information to .ssh/known_hosts. It uses the Windows registry. You can find details at https://superuser.com/questions/197489/where-does-putty-store-known-hosts-information-on-windows.

If you want to use AsyncSSH with known_hosts matching, you'll either need to run the Windows SSH client (which should be available on Windows 10) or manually edit the .ssh/known_hosts file to add an entry with the host's public key.

AsyncSSH also has a get_server_host_key() function you can use to get the host key, but you'd have to write a small amount of code to generate a known_hosts entry from that and append it to .ssh/known_hosts.

**Comment by Fanduzi** [comment]
try
```
asyncssh.connect('172.18.17.9', username="user", password="admin", port=9321, known_hosts=None) 
```

**Comment by ronf** [comment]
> try
>```python
> asyncssh.connect('172.18.17.9', username="user", password="admin", port=9321, known_hosts=None) 
> ```

While this would prevent you from needing to have a known hosts entry for the host, doing this would leave you wide open to man-in-the-middle attacks, so I wouldn't really recommend this.

> I don't know if asyncssh has a similar solution to using paramiko with the following two lines of code
> 
> ```python
> ssh.load_system_host_keys()
> ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
> ```

This also leaves you open to a MITM attack on the initial connection you make, and if you end up adding the attacker's host key instead of the real one, all future connections will be compromised as well. However, if you wanted to do something like this, AsyncSSH provides a couple of callbacks you can implement in the SSHClient class to do your own host key or host certificate authority validation. See #176 for more details.

Basically, you'd implement the `validate_host_public_key()` method in a subclass of SSHClient and have it store a local dictionary of keys it has seen, returning `True` if it had either already seen that host key or if it had no entry for that host yet, in which case you'd have it put the key passed to the callback into the dictionary. If the host was in the dictionary already with a different key, though, you'd have it fail the validation by returning `False`.

This callback is only called if no match is found among the loaded SSH host keys. So, it will still validate against anything found in `.ssh/known_hosts` first (if you don't specify the `known_hosts` argument), and only call your callback when no match was found.

**Comment by ronf** [comment]
Closing due to inactivity. Please open a new issue if you have additional questions.

**Comment by ialvata** [comment]
I'm actually having a similar problem. Spyder, a python IDE, uses asyncssh to create an ssh connection.

I'm using a Windows source to connect, through PuTTY, to a Linux target. When I'm using putty by itself, I know I can just accept the key on the first interaction, and then it'll be added to known hosts. The problem is that Spyder doesn't ask for this. It simply denies the connection.

To workaround this, I had to switch to using ssh in windows.
